addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -6543,182 +6553,247 @@ stock GetBossIndex(client)
+				switch(_operator[bracket])
 				{
-					case 1:
+					case Operation_Add:
 					{
-						sum[parentheses]+=playing;
+						sum[bracket] += sum[bracket+1];
+						break;
 					}
-					case 2:
+					case Operation_Subtract:
 					{
-						sum[parentheses]-=playing;
+						sum[bracket] -= sum[bracket+1];
+						break;
 					}
-					case 3:
+					case Operation_Multiplication:
 					{
-						sum[parentheses]*=playing;
+						sum[bracket] *= sum[bracket+1];
+						break;
 					}
-					case 4:
+					case Operation_Division:
 					{
-						sum[parentheses]/=playing;
+						sum[bracket] /= sum[bracket+1];
+						break;
 					}
-					case 5:
+					case Operation_Exponential:
 					{
-						sum[parentheses]=Pow(sum[parentheses], Float:playing);
+						sum[bracket] = Pow(sum[bracket], sum[bracket+1]);
+						break;
 					}
 					default:
 					{
-						parentheses=1;
+						//We don't got an operation for this ending bracket.(eg: "))")
+						sum[bracket] = sum[bracket+1];
 						break;
 					}
 				}
+				_operator[bracket+1] = Operation_None;
 			}
-
-			if(value[0]!='\0' && canAdd)
+			if (usePlayers)
+			{
+				usePlayers = false;
+				//Add sum with players.
+				switch(_operator[bracket])
+				{
+					case Operation_Add:
+					{
+						sum[bracket] += float(playing);
+						break;
+					}
+					case Operation_Subtract:
+					{
+						sum[bracket] -= float(playing);
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= float(playing);
+						break;
+					}
+					case Operation_Division:
+					{
+						sum[bracket] /= float(playing);
+						break;
+					}
+					case Operation_Exponential:
+					{
+						sum[bracket] = Pow(sum[bracket], float(playing));
+						break;
+					}
+					default:
+					{
+						sum[bracket] = float(playing);
+						break;
+					}
+				}
+			}
+			//TODO: Add useLives back.
+			if (strValue[0] != '\0') //Value has something in it, by time we get here we already have all the value parsed(hopefully).
 			{
-				canAdd=false;
-				switch(_operator[parentheses])
+				new Float:value = StringToFloat(strValue);
+				switch(_operator[bracket])
 				{
-					case 1:
+					case Operation_Add:
 					{
-						sum[parentheses]+=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] += value;
+						break;
 					}
-					case 2:
+					case Operation_Subtract:
 					{
-						sum[parentheses]-=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] -= value;
+						break;
 					}
-					case 3:
+					case Operation_Multiplication:
 					{
-						sum[parentheses]*=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] *= value;
+						break;
 					}
-					case 4:
+					case Operation_Division:
 					{
-						if(!StringToFloat(value))
+						if (!StringToInt(strValue) && StrContains(strValue, ".") == -1) //I can't find a good way to check if it's 0.0 due to things like 0.01 or 0.001
```

Comment:
then that would be we always say(dividing by 0.5 is invalid)

On Mon, Mar 16, 2015 at 8:38 PM, Wliu <notifications@github.com> wrote:

> In addons/sourcemod/scripting/freak_fortress_2.sp
> <https://github.com/50DKP/FF2-Official/pull/228#discussion_r26541120>:
>
> >  					{
> > -						if(!StringToFloat(value))
> > +						if (!StringToInt(strValue) && StrContains(strValue, ".") == -1) //I can't find a good way to check if it's 0.0 due to things like 0.01 or 0.001
>
> 0 is 0 no matter how you look at it WildCard ;P. We're not comparing
> floats here, we're comparing to see if it's equal to 0 or not. I'm pretty
> sure comparing to 0 will be *pretty* accurate.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/50DKP/FF2-Official/pull/228/files#r26541120>.
>